External server configuration

griffinhadfield opened the pull request:
Addressing https://github.com/OpenLiberty/guides-common/issues/540

Modification to use external server configuration

gkwan-ibm reviewed (changes requested):
@griffinhadfield Format the server.xml better for the following line:

![Screen Shot 2021-01-05 at 4 59 40 PM](https://user-images.githubusercontent.com/29576299/103704643-9e45a900-4f77-11eb-8207-578235f0b4c4.png)

gkwan-ibm reviewed (changes requested):
@griffinhadfield Update license year to 2021

and, apply to all your PRs for other guides

gkwan-ibm reviewed (changes requested):
@griffinhadfield The tests should not hard-coded "/dev/system/properties/". "/dev" should be configurable and pass through `-D` option when running `mvn failsafe:integration-test`

gkwan-ibm reviewed (changes requested):
@griffinhadfield should the start/system/src/main/liberty/config/server.xml hard-code the context root to `/`?

griffinhadfield commented:
> @griffinhadfield should the start/system/src/main/liberty/config/server.xml hard-code the context root to `/`?

@gkwan-ibm  You're right it should be hardcoded. I must have accidentally committed the server.xml after following the guide. I have fixed and pushed it.

gkwan-ibm reviewed (changes requested):
@griffinhadfield 
Now only 2 tests for SystemEndpointIT. The output in README need to be updated

AustinSeto reviewed (approved):
LGTM

Emily-Jiang commented:
@griffinhadfield for using kubernetes secret, normally you don't mount them as environment variables. You mount on a volume instead (see [here](https://github.com/OpenLiberty/open-liberty/issues/9786)). You can map to environment variables but it is not good practice. Is there a guide to cover k8s secret volume mounting and then get the variables injectable?

Emily-Jiang commented on README.adoc:
````diff
@@ -225,18 +229,18 @@ There are several ways to configure an environment variable in a Docker containe
 Create a Secret to configure the credentials that `inventory` will use to authenticate against `system` with the following `kubectl` command.
 [role='command']
 ```
 kubectl create secret generic sys-app-credentials --from-literal username=bob --from-literal password=bobpwd
````
The username and password should be different from the original. Otherwise, how do you know the new values were used. I suggest changing to username=alice password=wonderland or something different.

Emily-Jiang replied:
After you have the done the changes, you need to update the other places to pass in the right value.

Emily-Jiang reviewed (approved):
LGTM! This guide will be further improved after [this](https://github.com/OpenLiberty/open-liberty/issues/9786) is delivered.

Charlotte-Holt commented on README.adoc:
```diff
@@ -293,33 +315,30 @@ Run the following command to deploy your changes to the {kube} cluster.
-Navigate to `{system-api}` and examine the response headers in your developer console,
-or run the following `curl` command:
+Your application will now be available at the
+`http://[hostname]:31000/dev/system/properties`. You now need to use the new
```
Update

>Your application will now be available at the `http://[hostname]:31000/dev/system/properties`.

To

>Your application will now be available at the `http://[hostname]:31000/dev/system/properties` URL.

Charlotte-Holt commented on README.adoc:
```diff
@@ -330,23 +349,23 @@ include::{common-includes}/os-tabs.adoc[]
 [.tab_content.windows_section.mac_section]
 --
-Run the integration tests against a cluster running with appName `my-system`:
+Run the integration tests.
```
I'd put a colon after this to lead into the command.

Charlotte-Holt commented on README.adoc:
```diff
@@ -330,23 +349,23 @@ include::{common-includes}/os-tabs.adoc[]
 --
 
-The tests check that the `system` service responds with a header containing the configured name. The tests for `inventory` verify that the service may communicate with `system` using the configured credentials. If the credentials are misconfigured, then the `inventory` test will fail, so the `inventory` test indirectly verifies the credentials are correctly configured.
+The tests check that the `system` service is available at the correct context root. The tests for `inventory` verify that the service may communicate with `system` using the configured credentials. If the credentials are misconfigured, then the `inventory` test will fail, so the `inventory` test indirectly verifies the credentials are correctly configured.
```
Update

>The tests for `inventory` verify that the service may communicate with `system` using the configured credentials. If the credentials are misconfigured, then the `inventory` test will fail, so the `inventory` test indirectly verifies the credentials are correctly configured.

To

>The tests for `inventory` verify that the service can communicate with `system` using the configured credentials. If the credentials are misconfigured, then the `inventory` test fails, so the `inventory` test indirectly verifies that the credentials are correctly configured.

Charlotte-Holt commented on README.adoc:
````diff
@@ -225,18 +229,18 @@ There are several ways to configure an environment variable in a Docker containe
 ```
  
-This command looks very similar to the command to create a ConfigMap, one difference is the word `generic`. It means you're creating a Secret that is `generic`, in other words it stores information that isn't specialized in any way. There are different types of secrets, such as secrets to store Docker credentials and secrets to store public and private key pairs.
+This command looks similar to the command to create a ConfigMap, one difference is the word `generic`. This creates a Secret that doesn't store information in any specialized way. There are different types of secrets, such as secrets to store Docker credentials and secrets to store public and private key pairs.
````
Update

>This command looks similar to the command to create a ConfigMap, one difference is the word `generic`. This creates a Secret that doesn't store information in any specialized way.

To

>This command looks similar to the command to create a ConfigMap, but one difference is the word `generic`. This word creates a Secret that doesn't store information in any specialized way.

Charlotte-Holt commented on README.adoc:
```diff
@@ -270,7 +272,27 @@ kubernetes.yaml
+file=0]`secretKeyRef` gets the values [hotspot=username1 hotspot=username2
+file=0]`username` and [hotspot=password1 hotspot=password2 file=0]`password`
+from the Secret [hotspot=credentials1 hotspot=credentials2 hotspot=credentials3
+hotspot=credentials4 file=0]`sys-app-credentials`.
```
Update to

>Using the `valueFrom` field, you can specify the value of an environment variable from various sources. These sources include a ConfigMap, a Secret, and information about the cluster. In this example `configMapKeyRef` gets the value `contextRoot` from the `sys-app-root` ConfigMap. Similarly, `secretKeyRef` gets the values `username` and `password` from the `sys-app-credentials` Secret.

Charlotte-Holt commented on README.adoc:
```diff
@@ -270,7 +272,27 @@ kubernetes.yaml
+[hotspot=contextRoot1 hotspot=contextRoot2 file=0]`CONTEXT_ROOT`,
+[hotspot=sysUsername1 hotspot=sysUsername2 file=0]`SYSTEM_APP_USERNAME`, and
+[hotspot=sysPassword1 hotspot=sysPassword2 file=0]`SYSTEM_APP_PASSWORD` environment
+variables are set.
```
Update to

>The `CONTEXT_ROOT`, `SYSTEM_APP_USERNAME`, and `SYSTEM_APP_PASSWORD` environment variables are set in the `env` sections of `system-container` and `inventory-container`.

Charlotte-Holt reviewed (commented):
@griffinhadfield I added ID review feedback. Let me know if you have any questions.

griffinhadfield commented:
@Charlotte-Holt Thank you for the feedback. I have addressed your comments, the updated README can be see here: http://lgdev1.fyre.ibm.com:4020/guides/kubernetes-microprofile-config
